addons/certificate/models/certificate.py

```diff
@@ -0,0 +1,296 @@
+        string='Public Key',
+        comodel_name='certificate.key',
+        check_company=True,
+        domain=[('public', '=', True)],
+        help="""Used to set a public key in case the one self-contained in the certificate is erroneus.
+                When a public key is set this way, it will be used instead of the one in the certificate.
+             """,
+    )
+    scope = fields.Selection(
+        string="Certificate scope",
+        selection=[
+            ('general', 'General'),
+        ],
+    )
+    content_format = fields.Selection(
+        selection=[
+            ('der', 'DER'),
+            ('pem', 'PEM'),
+            ('pkcs12', 'PKCS12'),
+        ],
+        string='Original certificate format',
+        store=True,
+        compute='_compute_pem_certificate',
+    )
+    pem_certificate = fields.Binary(
+        string='Certificate in PEM format',
+        attachment=False,
+        store=True,
+        compute='_compute_pem_certificate',
+    )
+    serial_number = fields.Char(
+        string='Serial number',
+        help='The serial number to add to electronic documents',
+        compute='_compute_pem_certificate',
+        store=True,
+    )
+    date_start = fields.Datetime(
+        string='Available date',
+        help='The date on which the certificate starts to be valid (UTC)',
+        compute='_compute_pem_certificate',
+        store=True,
+    )
+    date_end = fields.Datetime(
+        string='Expiration date',
+        help='The date on which the certificate expires (UTC)',
+        compute='_compute_pem_certificate',
+        store=True,
+    )
+    is_valid = fields.Boolean(string='Valid', compute='_compute_is_valid', search='_search_is_valid')
+    active = fields.Boolean(name='Active', help='Set active to false to archive the certificate', default=True)
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        required=True,
+        default=lambda self: self.env.company,
+        ondelete='cascade',
+    )
+    country_code = fields.Char(related='company_id.country_id.code', depends=['company_id'])
+
+    # Technical field used to display a certificate loading error
+    loading_error = fields.Text(string='Loading error', compute="_compute_pem_certificate", store=True)
+
+    @api.depends('content', 'pkcs12_password')
+    def _compute_pem_certificate(self):
+        for certificate in self:
+            content = certificate.with_context(bin_size=False).content
+            if not content:
+                certificate.pem_certificate = None
+                certificate.date_start = None
+                certificate.date_end = None
+                certificate.serial_number = None
+                certificate.private_key = None
+                certificate.public_key = None
+                certificate.loading_error = ""
+                certificate.scope = certificate.scope or certificate.env.context.get('scope', 'general')
```

Comment:
That's really a bad design to modify the behavior of a computed stored field depending a context key.